src/test/java/io/vertx/json/schema/CBORTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2011-2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.json.schema;
+
+import com.fasterxml.jackson.core.Base64Variants;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonTokenId;
+import com.fasterxml.jackson.dataformat.cbor.CBORFactory;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.VertxExtension;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.IOException;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+@ExtendWith(VertxExtension.class)
+public class CBORTest {
+
+  private static final Base64.Encoder B64ENC = Base64.getUrlEncoder().withoutPadding();
+  private static final Base64.Decoder B64DEC = Base64.getUrlDecoder();
+  private static final CBORFactory FACTORY = new CBORFactory();
+
+  @Test
+  public void schemaAndValidationUsingCBOR(Vertx vertx) throws IOException {
+
+    SchemaRouter router = SchemaRouter.create(vertx, new SchemaRouterOptions());
+    SchemaParser parser = SchemaParser.createDraft7SchemaParser(router);
+
+    Schema schema = parser.parse(
+      new JsonObject()
+        .put("type", "object")
+        .put("required", new JsonArray().add("fmt").add("authData").add("attStmt")));
+
+    String data = "o2NmbXRkbm9uZWdhdHRTdG10oGhhdXRoRGF0YVjEfxV8VVBPmz66RLzscHpg5yjRhO28Y_fPwYO5AVwzBEJBAAAAAwAAAAAAAAAAAAAAAAAAAAAAQEPjBz9F6ttAijOS1t6gbfYLub3TmCzWXN4wnIMsi53EWr-SL3e09XWr93lqyOwk_B5s1P8gGCa5o2uIp_DhS9ylAQIDJiABIVggN_D3u-03a0GzONOHfaML881QZtOCc5oTNRB2wlyqUEUiWCD3878XoO_bIJf0mEPDILODFhVmkc4QeR6hOIDvwvXzYQ";
```

Comment:
How about writing this as json, convert to cbor and then reconvert back to map?

Reply:
I used a dump CBOR object (from webauthn) because I wanted to keep the test dependencies to the minimum, if we're going to do encode/decode, then I should also pull `databind` as a dependency and remove the handcrafted simple parser as writing parsers is not really the point of the test.